### DISCUSSION TITLE
PGlite can't sync from electric server (always showing an empty array)

### DISCUSSION THREAD

**Discussion by mfaridzia** [electric-sql/pglite#254] [Q&A] [answered]
Hey @samwillis before, thank you for amazing PGlite 🙏

I have a problem, I am trying to sync my electric server data to my local table in PGlite, but it's always showing an empty array. I don't have any idea where the problem is.

<img width="698" alt="image" src="https://github.com/user-attachments/assets/fd5df220-e1b6-4c22-a963-da88ef4b0b7a">

This is my GitHub repo: https://github.com/mfaridzia/electric/blob/main/src/PgLite2.tsx

Thank you.

**Comment by TWith2Sugars** [comment]
I had this problem as well, I believe the latest version of electric has changed some properties of the return types:
when I updated `action` (as shown here https://github.com/electric-sql/pglite/blob/main/packages/pglite-sync/src/index.ts#L159) to operation it worked

**Reply by mfaridzia** [reply]
I see. When I saw the log, it used _operation_ instead of _action_.
<img width="585" alt="image" src="https://github.com/user-attachments/assets/073d58f1-52a8-430c-85ef-a58c3f7031fc">

But I still wonder how I can change the _headers.action_ to _headers.operation_ @TWith2Sugars ? Thank you.

**Reply by TWith2Sugars** [reply]
I ended up grabbing a copy of the file and modified it locally.

**Comment by samwillis** [comment]
Hey @mfaridzia @TWith2Sugars 

Well spotted, we haven't updated the plugin to match this change to the sync server: https://github.com/electric-sql/electric/pull/1481

@balegas may be a good one for you to take a look at as you wanted to work on the sync plugin?

tracking issue here: https://github.com/electric-sql/pglite/issues/260

**Comment by msfstef** [comment, marked as answer]
Fixed by https://github.com/electric-sql/pglite/pull/263
